initialization/initialization_test.go

```diff
@@ -595,9 +600,16 @@ func TestInitialize_MultipleFiles(t *testing.T) {
 			}
 			r.NoError(verifying.VerifyVRFNonce(init.Nonce(), m))
 
-			// TODO(mafa): see above
-			// r.Equal(oneFileNonce, *init.Nonce())
+			if nonce == nil {
```

Comment:
If you handle `numFiles = 1` as a special case for reference. And then increase `numFiles` for every subsequent iteration, this handling of `nonce == nil` becomes unnecessary. The same is true for `data` / `lastData`.

Also this nonce is now never verified to be valid (before it was with: `r.NoError(verifying.VerifyVRFNonce(init.Nonce(), m))`)

Reply:
> If you handle `numFiles = 1` as a special case for reference. And then increase `numFiles` for every subsequent iteration, this handling of `nonce == nil` becomes unnecessary. The same is true for `data` / `lastData`.

I made this change because I thought that handling the `nonce == nil` case is much more straightforward than handling the reference case. 

> Also this nonce is now never verified to be valid (before it was with: `r.NoError(verifying.VerifyVRFNonce(init.Nonce(), m))`)

It's still being verified using the same code.

Reply:
> I made this change because I thought that handling the nonce == nil case is much more straightforward than handling the reference case.

It depends. I think reducing nesting is better for readability than avoiding code completion:

```golang
var nonce *int
var data []byte

for numFiles := uint(1); numFiles <= 16; numFiles <<= 1 {
	t.Run(fmt.Sprintf("NumFiles=%d", numFiles), func(t *testing.T) {
		if data == nil {
			// only applies to the first loop
			data = lastData
		} else {
			// applies to all other loops
			r.Equal(data, lastData)
		}

		if none == nil {
			// only applies to the first loop
			*nonce = *init.Nonce()
		} else {
			// applies to all other loops
			r.Equal(*nonce, *init.Nonce())
		}
	}
}
```

is much harder to read than:

```golang
nonce := init.Nonce()
data := generateDate()

for numFiles := uint(2); numFiles <= 16; numFiles <<= 1 {
	t.Run(fmt.Sprintf("NumFiles=%d", numFiles), func(t *testing.T) {
		r.Equal(data, lastData)

		r.Equal(*nonce, *init.Nonce())
	}
}
```

Reply:
> It's still being verified using the same code.

Ah, my bad I misread the diff 😓

Reply:
I still think it's currently simpler to read when considering the code duplication and having value 2 as the starting point.. but I can fix it if you want 😉

Reply:
That would be great ❤️  This would also fix (some of) the issues reported by `staticcheck`.